src/generator/docker-compose-file-generator.ts

```diff
@@ -1,40 +1,165 @@
-import getInitialDockerComposeFile from "../helpers/initial-docker-compose-file.js";
+import { AbstractFileGenerator } from "./file-generator.js";
 
 interface LiveUpdate {
   liveUpdate: boolean;
 }
 
 type ServiceWithLiveUpdate = Service & LiveUpdate;
 
+/**
+ * Class responsible for generating Docker Compose files based on service configurations.
+ */
 export class DockerComposeFileGenerator extends AbstractFileGenerator {
+    /**
+     * Constructor for DockerComposeFileGenerator.
+     * @param path - The path to the directory where the Docker Compose root file will be created.
+     */
     constructor (path: string) {
         super(path, "docker-compose.yaml");
     }
 
-    generateDockerComposeFile (services: ServiceWithLiveUpdate[], excluded: string[] | undefined) {
-        const dockerCompose = getInitialDockerComposeFile(this.path);
-
-        // Remove any excluded services (if supplied)
-        const runnableServices = typeof excluded !== "undefined"
+    /**
+     * Removes excluded services from the list of services.
+     * @param services - List of all services with live updates.
+     * @param excluded - List of service names to exclude.
+     * @returns Filtered list of services.
+     */
+    private removeExcludedServices (services: ServiceWithLiveUpdate[], excluded?: string[] | undefined):ServiceWithLiveUpdate[] {
+        return typeof excluded !== "undefined"
             ? services.filter(service => !excluded.includes(service.name))
             : services;
+    }
+
+    /**
+     * Adds services to the `include` property of the main Docker Compose configuration.
+     * Handles live updates and excluded services.
+     * The development enabled services docker-compose.yaml file in local repository are linked into the include property instead of  new docker-compose.yaml file will not be generated.
+     * @param dockerCompose - Docker Compose configuration object.
+     * @param runnableServices - List of services to include.
+     * @param hasExcludedServices - Indicates if there are excluded services.
+     * @returns Updated Docker Compose configuration.
+     */
+    private addIncludePropertiesToMainDockerCompose (dockerCompose:DockerComposeSpec, runnableServices:ServiceWithLiveUpdate[], hasExcludedServices = false): DockerComposeSpec {
+        if (!hasExcludedServices) {
+            dockerCompose.include = runnableServices.map(
+                service => service.liveUpdate
+                    ? join(this.path, "local", service.name, "docker-compose.yaml")
+                    : service.source
+            ).reduce(deduplicate, []);
+        } else {
+            const includes: string[] = [];
+            runnableServices.forEach(service => {
+                if (service.liveUpdate) {
+                    includes.push(join(this.path, "local", service.name, "docker-compose.yaml"));
+                }
+            });
```

Comment:
I prefer `filter`s/`map`s - less mutable code. But that's my personal preference - feel free to disagree if you're not keen.